src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -199,12 +199,25 @@ def execute(self, args):
                                   EVENT_INVOKER_FILTER_RESULT)
         from knack.util import CommandResultItem, todict
         from azure.cli.core.commands.events import EVENT_INVOKER_PRE_CMD_TBL_TRUNCATE
+        from azure.cli.core.commands import cache
 
         # TODO: Can't simply be invoked as an event because args are transformed
         args = _pre_command_table_create(self.cli_ctx, args)
 
         self.cli_ctx.raise_event(EVENT_INVOKER_PRE_CMD_TBL_CREATE, args=args)
-        self.commands_loader.load_command_table(args)
+        success = False
+        if cache.cache_exists():
```

Comment:
This logic doesn't make sense. Why does it try to use the cache and then only later ask whether the user actually wants to use the cache?  Also, it just not just poll the raw ENV var--it should look in the config file. That will automatically check the config and the ENV VAR.

Reply:
Doing this will also allow the user to persist in their config file that so that they opt-in to this behavior (which is appropriate).